### DISCUSSION TITLE
Background syncing

### DISCUSSION THREAD

**Discussion by melursus23** [ScottArbeit/Grace#13] [General]
Hello

I have some questions about the "auto" branch syncing happening in the background, if I understood correctly.

1. How will you deal when someone is trying to troubleshoot a bug. If I'm in that situation, I do not want anything to be sync or changed locally since it could interfered with my troubleshooting session?
2. It's not because I'm able to sync code remotely with my local changes without merge conflict that the code will compile, how will you deal with that?

**Comment by ScottArbeit** [comment]
Hey @melursus23!

I agree that there are going to be temporary circumstances where you don't want auto-rebase to happen. And there are going to be work situations where it's not appropriate to have auto-rebase turned on at all. There are edge cases here that I'm sure I haven't thought of yet. (There always are.)

Grace will absolutely have to have the ability to turn off auto-rebase temporarily and/or permanently in a repository. I think it also will need a mode where it presents a prompt of some sort that an auto-rebase is available, and should we do it? I haven't written any of that code yet, but I'm totally aware that it needs to be there.

With that said, I think it's important to make auto-rebasing the default behavior of Grace. When I think about my years using Git, and how many times I've rebased my branch on the current state of `master` or `main`, and how many times that has caused me a problem - maybe once or twice? definitely less than five - I think most developers will find it safe to have on. I mean, I hope people haven't had too many problems getting their branch up-to-date in Git in general. I'm not 100% about this, but I kind-of want to say: if you show me a dev team that has massive, ongoing problems with rebasing I'll show you some poor dev practices, or some code files that need to be broken up.

I believe that auto-rebasing will help to prevent more conflicts than it will cause confusion. There will need to be some sort of notification mechanism that an auto-rebase has happened, so you feel like you're in control of what's happening. My hope is that having it happen automatically will turn out to be absolutely fine 95% of the time, and the other 5% will be 4% conflicts and 1% "oops something bad happened that I have to debug". If we come close to that ratio, the feature will be successful at making our lives easier.
